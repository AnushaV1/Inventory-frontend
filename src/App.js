import React, {useState, useEffect} from "react";
import {useDispatch} from "react-redux";
import Alerts from './components/Alerts';
import {decode} from 'jsonwebtoken';
import {  loadUser } from './actions/userAction';
import './App.css';
import Navigation from "./components/Navigation";
import Routes from "./components/Routes";
import Footer from "./components/Footer";
import { loadAllUPC } from "./actions/upcActions";

function App() {
  const dispatch = useDispatch();

  const [infoLoaded, setInfoLoaded] = useState(false);

    useEffect(() => {
    async function getUser() {
      let token = localStorage.getItem('inventory-token') || null;
              if(token) {
                let { username,userid } = decode(token);
                let data = {token, username, userid}
                console.log("Data from app js", data)
                await dispatch(loadUser(data));
                await dispatch(loadAllUPC());
          }
    setInfoLoaded(true);
    }
    getUser();
    }, [dispatch]);



  if (!infoLoaded) {
    return <h3>Loading...</h3>;
  }


  return (

        <div className="App">
              <Navigation />
              <Alerts />
              <Routes  />
              <Footer />
              </div>
  )
}

export default App;
