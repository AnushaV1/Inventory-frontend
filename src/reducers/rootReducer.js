import { combineReducers } from 'redux';
import { users }  from "./authReducer";
import  { alert }  from "./alertReducer";
import  { upc }  from "./upcReducer";
import { products } from "./userInventoryReducer"

const rootReducer = combineReducers({

    users,
    upc,
    alert,
    products
});

export default rootReducer;