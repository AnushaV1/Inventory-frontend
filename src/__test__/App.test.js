import React from 'react';
import { render } from '@testing-library/react';
import { Provider } from 'react-redux';
import rootReducer from '../reducers/rootReducer';
import { createStore } from 'redux';
import { BrowserRouter } from 'react-router-dom';
import App from '../App';

const store = createStore(rootReducer);

it('renders without crashing', function () {
	render(
		<Provider store={store}>
			<BrowserRouter>
				<App />
			</BrowserRouter>
		</Provider>
	);
});

it('matches snapshot', function () {
	const { asFragment } = render(
		<Provider store={store}>
			<BrowserRouter>
				<App />
			</BrowserRouter>
		</Provider>
	);
	expect(asFragment()).toMatchSnapshot();
});