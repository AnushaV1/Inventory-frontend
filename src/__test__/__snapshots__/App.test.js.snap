// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="App"
  >
    <nav
      class="navbar navbar-expand-md navbar-dark bg-dark"
    >
      <div
        class="container-fluid"
      >
        <div
          class="navbar-header"
        >
          <a
            aria-current="page"
            class="navbar-brand active"
            href="/"
          >
            <img
              alt="Home Inventory Logo"
              src="homeInventoryLogo.jpg"
            />
            My Home Inventory
          </a>
        </div>
        <ul
          class="nav navbar-nav navbar-right"
        >
          <li
            class="nav-item"
          >
            <a
              class="nav-link"
              href="/register"
            >
              Register
            </a>
          </li>
          <li
            class="nav-item"
          >
            <a
              class="nav-link"
              href="/login"
            >
              Login
            </a>
          </li>
        </ul>
      </div>
    </nav>
    <div />
    <div>
      <section
        class="Home col-md-12 p-0"
      >
        <div
          class="banner container-fluid text-center text-light m-0 p-5"
        >
          <div
            class="content-wrapper mb-5"
          >
            <h1
              class="mb-3 font-weight-bold"
            >
              Inventory
            </h1>
            <p
              class="lead"
            >
              All your inventory in one, convenient place.
            </p>
            <a
              class="btn btn-primary btn-lg px-5 py-3 my-5"
              href="/login"
            >
              Log In
            </a>
          </div>
        </div>
      </section>
    </div>
    <footer
      class="footer"
    >
      <div
        class="container-fluid text-center"
      >
        <div
          class="row"
        >
          <div
            class="col-md-4 mt-md-0 mt-3 text-left"
          >
            <h5>
              <a
                class="text-white-50"
                href="https://www.upcitemdb.com/api/explorer#!/lookup/get_trial_lookup"
              >
                API - UPCItemDB 
              </a>
            </h5>
          </div>
          <div
            class="col-md-4 mb-md-0 mb-3 "
          >
            <h5
              class="text-white-50"
            >
              Springboard Capstone project 2
            </h5>
          </div>
          <div
            class="col-md-4 mb-md-0 mb-3 text-center"
          >
            <h4>
               
              <a
                class="text-white-50"
                href="http://www.linkedin.com/in/anusha-venkataraghavan"
              >
                <i
                  class="fab fa-linkedin"
                />
              </a>
            </h4>
          </div>
        </div>
      </div>
    </footer>
  </div>
</DocumentFragment>
`;
